pkg/pillar/pubsub/subscribe.go

```diff
@@ -50,6 +51,10 @@ func (sub *SubscriptionImpl) MsgChan() <-chan Change {
 
 // Activate starts the subscription
 func (sub *SubscriptionImpl) Activate() error {
+	if sub.activated {
+		return errors.New("already activated")
+	}
+	sub.activated = true
```

Comment:
Should we set false in Close() function?

Reply:
Right, we definitely should - done